### PR TITLE
#928 Fix various small bugs in config processing

### PR DESCRIPTION
**Pull request by rmjarvis**
This PR includes a number of small bug fixes that @NiallMac and I ran into when trying to do some recent DES simulations.

1. The des_meds maker wasn't setting the position of the object within the stamp correctly.  It was just giving the center of the stamp, not including any sub-pixel offset.  This is now fixed.
2. The FocalPlane type (in the galsim_extra repo) uses multiple rng cadences to get some values to repeat for each chip in an exposure.  This required the seed dict for one of them to know the number of objects in the exposure.  But if the number of objects itself was a random variate, this led to a chicken and egg problem where the requisite rng wasn't set before it was needed.  The solution is to set the first rng in the list before parsing later ones.
3. FocalPlane needs to do some special setup before `SetupConfigRNG` is called.  There used to be a hacky workaround to make this kind of work.  But it was buggy, and the easiest fix seemed to be to just add an OutputBuilder function called `setup` that gets called at the start of building a file.  The default behavior for this function is just to call `SetupConfigRNG`, but it provides a non-hacky place for FocalPlane to do its extra setup stuff, which now seems to work correctly.
4. There was a very subtle bug in how the Eval code was working.  When checking which `eval_varaibles` to load up, it used to just check `if key in str` to see which keys were somewhere in the eval string.  But this found shorter words than were intended.  E.g. `focal_r` when `focal_rmax` was in the string.  This led to `focal_r` being evaluated as well as `focal_rmax`.  Normally this would be fine, but if this happened at image-processing scope rather than object-processing scope, the wrong rng could be used, which would not be re-calculated for the first object in an image, since the code would think it was already current.  TL;DR we now check for the whole word in the string using regular expressions.
5. There was a minor bug in SetupDefaultIndex where a non-default index_key wasn't being correctly propagated into the new index field.
6. Normally `world_center` is just calculated as the world position of the image center.  This is used whenever we need to do a tangent-plane projection of the world coordinates.  But FocalPlane uses the exposure center instead.  It had been using secret knowledge of GalSim's config code to effect this, but here I made this a real parameter in the `image` field that users can set, which seems like a better hook to provide.
7. The `eval_base_variables` list is now imported into the galsim.config namespace to make it more convenient for the user to add to.

### CODE REVIEW

**Comment by rmjarvis** [comment]
@NiallMac I think with this branch for GalSim and the focal_plane branch in galsim_extra, everything seems to be working.  Please check it out when you get a chance.

File: galsim/config/output.py [hunk 1]
```diff
@@ -418,6 +415,20 @@ class OutputBuilder(object):
     # A class attribute that sub-classes may override.
     default_ext = '.fits'
 
+    def setup(self, config, base, file_num, logger):
+        """Do any necessary setup at the start of processing a file)
```
Comment (rmandelb) [inline galsim/config/output.py]: random parenthesis

File: galsim/config/process.py [hunk 2]
```diff
@@ -1019,21 +1031,22 @@ def GetIndex(config, base, is_sequence=False):
...
     """Get the appropriate current rng according to whatever the current index_key is.
 
     If a logger is provided, then it will emit a warning if there is no current rng setup.
 
     @param config           The configuration dict for the current item being worked on.
     @param base             The base configuration dict.
     @param logger           If given, a logger object to log progress. [default: None]
-    @param tag              If given, an appropriate name for the current item to use ing the
+    @param tag              If given, an appropriate name for the current item to use in the
                             warning message. [default: None]
```
Comment (rmandelb) [inline galsim/config/process.py]: default: '' (not None)

File: galsim/config/process.py [hunk 3]
```diff
@@ -1046,14 +1059,33 @@ def GetRNG(config, base, logger=None, tag=None):
...
 
+def CleanConfig(config): # pragma: no cover
+    """Return a "clean" config dict without any leading-underscore values
+
+    GalSim config dicts store a lot of ancillary information internally to help improve
+    efficiency.  However, some of these are actually pointers to other places in the dict, so
+    printing a config dict, or even what should be a small portion of one, can have infinite loops.
+
+    This helper function is useful when debugging config processing to strip out all of these
+    leading-underscore values, so that printing the dict is reasonable.
```
Comment (rmandelb) [inline galsim/config/process.py]: Nice.  I will definitely appreciate this when debugging config-related issues.

File: tests/test_des.py [hunk 4]
```diff
@@ -267,6 +267,13 @@ def test_meds_config():
     g1 = -0.17
     g2 = 0.23
 
+    # generate offsets that depend on the on the object num so can be easily reproduced
```
Comment (rmandelb) [inline tests/test_des.py]: repeats "on the on the"

**Comment by rmjarvis** [comment]
Thanks Rachel.  Typos fixed.

**Comment by NiallMac** [comment]
All looks good to me, thanks both! 

I'll have a think about some more FocalPlane unit tests that test stuff at the image level as well as the truth catalogs...

**Pull request merged**